src/QMCWaveFunctions/TWFFastDerivWrapper.h

```diff
@@ -57,6 +61,28 @@ class TWFFastDerivWrapper
   void addGroup(const ParticleSet& P, const IndexType groupid, SPOSet* spo);
   inline void addJastrow(WaveFunctionComponent* j) { jastrow_list_.push_back(j); };
 
+  /**
+   * @brief add a MultiSlaterDet
+   * 
+   * this is called when registering a MSD
+   * the caller should also iterate over MultiDiracDeterminants in the MSD and register the associated SPOSets
+   * 
+   * @param P. ParticleSet 
+   * @param msd. Pointer to MultiSlaterDetTableMethod
+   */
+  void addMultiSlaterDet(const ParticleSet& P, const WaveFunctionComponent* msd);
+  inline IndexType numMultiSlaterDets() const { return slaterdets_.size(); }
+  inline bool hasMultiSlaterDet() const
+  {
+    if (numMultiSlaterDets() > 1)
+    {
+      APP_ABORT("ERROR: TWFFastDerivWrapper does not support multiple MultiSlaterDetTableMethod");
+    }
+    return (numMultiSlaterDets() > 0);
+  }
+  const WaveFunctionComponent& getMultiSlaterDet(const IndexType id) const { return *slaterdets_[id]; }
```

Comment:
Confused. I thought there is only one MultiSlaterDet allowed. What does this function do?

Reply:
There is only one allowed for now; if there's any use case for more (like a product of MSD terms), I could extend this to handle that case.
This seemed like the easiest way to handle checking whether we've added more than one (which is not currently supported by the wrapper), while also allowing for that to be implemented in the future with minimal changes to the wrapper class data members.

An alternative would be to just keep a single shared_ptr<const WaveFunctionComponent> (that defaults to nullptr, and stays nullptr while this wrapper doesn't have a MSD). Then the checks like  `if(slaterdets_.size() > 0)` can just be changed to `if(slaterdets_)`.

I can change it to `shared_ptr<const WaveFunctionComponent> slaterdet_;` if you agree

Reply:
No need to change the code, just document this function and mention currently only support 1 multidet.

Reply:
I changed it so that it only has `const WaveFunctionComponent* slaterdet_;`

I think the MSD should stay in scope for as long as we need it here (if we wanted to be safer, we could use [enable_shared_from_this](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this) in the MSD class and make it a shared_ptr)

Reply:
I made the comment before seeing your changing. Your change is good. No need to revert.